docs/src/atomistic/overview.rst

```diff
@@ -0,0 +1,268 @@
+computed by the model follow the metatensor interface, which can be used to
+debug your code.
+
+Finally, the exported model can be loaded by simulation engines and used to run
+simulations and make predictions.
+
+
+.. _TorchScript: https://pytorch.org/docs/stable/jit.html
+
+Constrains on atomistic models
+------------------------------
+
+There are a couple of constrains on what a given model must do to be useable
+with metatensor, but appart from these you can do what you want inside the
+model!
+
+The main constrain is that the model must be compatible with `TorchScript`_,
+i.e. you must use either pure PyTorch code in the definition of your model, or
+implement a custom TorchScript extension for any operations where a pure PyTorch
+implementation is too slow or too much work.
+
+Your model should also only take data from our :py:class:`System` definition:
+atomic types and positions, simulation cell, and list of neighbors for different
+spherical cutoffs.
+
+If you need additional data that you can not compute inside the model (such as
+atomic spins, non-spherical particle orientation, *etc.*) you can use
+:py:meth:`System.get_data`, but this comes with significant caveats:
+
+- anything going through :py:meth:`System.get_data` is experimental with no
+  stability guarantee;
+- you must modify the engine code to use :py:meth:`System.add_data` to add the
+  required data to the systems;
+
+If you need such data, please contact us to formulate a plan to add it to
+metatensor interface!
+
+Finally, your model can compute and output what it wants, and organize the data
+and metadata of the outputs as it pleases, except for a set of standardized
+outputs (identified by the corresponding key in the output dictionary). These
+standardized outputs are documented in :ref:`this page
+<atomistic-models-outputs>`.
+
+
+Data flow between the model and engine
+--------------------------------------
+
+The sequence of operations to use a metatensor atomistic model from a simulation
+engine follows the same high level sequence of operations, illustrated and
+explained below.
+
+
+.. make the `tip` admonition grey only for this page
+.. raw:: html
+
+    <style>
+        body[data-theme="light"] {
+            --color-admonition-title--tip: #7c7c7c;
+            --color-admonition-title-background--tip: #b9b9b9;
+        }
+
+        body[data-theme="auto"] {
+            @media (prefers-color-scheme: light) {
+                --color-admonition-title--tip: #7c7c7c;
+                --color-admonition-title-background--tip: #b9b9b9;
+            }
+        }
+    </style>
+
+.. figure:: ../../static/images/model-dataflow.*
+    :width: 600px
+    :align: center
+
+    Illustration of the flow of data between the engine and the model.
+
+1. the engine loads an exported model from a file;
+
+   .. tip::
+
+        The engine should use :py:func:`check_atomistic_model` or
+        :cpp:func:`metatensor_torch::load_atomistic_model` to also perform
```

Comment:
Is this also exposed to Python? We might also use this in metatensor-models. ☺️

Reply:
Yes, it is exposed to Python! https://lab-cosmo.github.io/metatensor/latest/atomistic/reference/models/index.html#metatensor.torch.atomistic.check_atomistic_model